roles/st2chatops/tasks/main.yml

```diff
@@ -26,18 +27,55 @@
   when: st2chatops_version != "latest"
   tags: [st2chatops, skip_ansible_lint]
 
-- name: Add st2_api_key in st2chatops.env
+- name: Check if api key already exist. No change if it already exists
+  become: yes
+  shell: "grep -e '^export ST2_API_KEY=\"\\${ST2_API_KEY}\"$' /opt/stackstorm/chatops/st2chatops.env"
+  changed_when: no
+  ignore_errors: True
+  register: task_apikey_exists
+
+- name: Add user defined "st2chatops_st2_api_key" in st2chatops.env, from "role" or "defaults/main.yml"
   become: yes
   replace:
     dest: /opt/stackstorm/chatops/st2chatops.env
-    regexp: '(?<=ST2_API_KEY=)(.*)$'
+    regexp: '(?<=ST2_API_KEY=)(\"\$\{ST2_API_KEY\}\")$'
     replace: '{{ st2chatops_st2_api_key }}'
-  when: st2chatops_st2_api_key != 'CHANGE-ME-PLEASE'
-  register: task_st2_api_key
+  when: >
+    (task_apikey_exists.failed is not defined) and
+    (st2chatops_st2_api_key is defined) and
+    (st2chatops_st2_api_key != "CHANGE-ME-PLEASE")
+  register: task_user_st2_api_key
   no_log: true
   notify: restart st2chatops
   tags: st2chatops
 
+- name: Generate authentication token
+  command: st2 auth {{ st2_auth_username }} -p {{ st2_auth_password }} -t
+  when: (task_apikey_exists.failed is not defined) and (task_user_st2_api_key.changed == false)
```

Comment:
> Otherwise it means that st2chatops role included without st2 role.

We may not hit this until someone deliberately removes `st2` role from `st2chatops` dependencies (even if it is not included in `stackstorm.yml` playbook).. We don't take responsibility if a user edits the meta/tasks unnecessarily. Right now if you provide api_key with the role it will take precedence over generate api task and last one in the line is default.

In case we plan to remove `st2` role as a dependency, we are not handling it and we know where to fix.